src/lib/stream/gst/utils.rs

```diff
@@ -110,32 +111,63 @@ pub async fn wait_for_element_state_async(
 }
 
 #[instrument(level = "debug")]
-pub async fn get_encode_from_rtspsrc(stream_uri: &url::Url) -> Option<VideoEncodeType> {
+pub async fn get_encode_from_stream_uri(stream_uri: &url::Url) -> Option<VideoEncodeType> {
     use gst::prelude::*;
 
-    let description = format!(
-        concat!(
-            "rtspsrc location={location} is-live=true latency=0",
-            " ! fakesink name=fakesink sync=false"
-        ),
-        location = stream_uri.to_string(),
-    );
+    let description = match stream_uri.scheme() {
+        "rtsp" => {
+            format!(
+                concat!(
+                    "rtspsrc location={location} is-live=true latency=0",
+                    " ! typefind name=typefinder minimum=1",
+                    " ! fakesink name=fakesink sync=false"
+                ),
+                location = stream_uri.to_string(),
+            )
+        }
+        "udp" => {
+            format!(
+                concat!(
+                    "udpsrc address={address} port={port} close-socket=false auto-multicast=true",
+                    " ! typefind name=typefinder minimum=1",
+                    " ! fakesink name=fakesink sync=false"
+                ),
+                address = stream_uri.host()?,
+                port = stream_uri.port()?,
+            )
+        }
+        unsupported => {
+            warn!("Scheme {unsupported:#?} is not supported for Redirect Pipelines");
+            return None;
+        }
+    };
 
     let pipeline = gst::parse::launch(&description)
         .expect("Failed to create pipeline")
         .downcast::<gst::Pipeline>()
         .expect("Pipeline is not a valid gst::Pipeline");
 
+    let typefinder = pipeline.by_name("typefinder")?;
+
+    let (tx, rx) = mpsc::channel(10);
```

Comment:
A oneshot makes more sense here, no ?

Reply:
That was my first try, but oneshot requires T to impl copy  :(